example/App.tsx

```diff
@@ -34,51 +44,122 @@ const styles = StyleSheet.create({
+                Geolocation.checkIOSAccuracyAuthorization().then((acc: AccuracyAuthorization) => {
+                    setAccuracyAuthorization(acc);
+                });
+            } else {
+                /* Checking fine location permission will give us the information on whether the user has activated precise
+                location or not */
+                PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION).then((isAuthorized) => {
+                    const acc = isAuthorized ? 'FullAccuracy' : 'ReducedAccuracy';
+                    setAccuracyAuthorization(acc);
+                });
+            }
+        };
+
+        const handleAppStateChange = (nextAppState: AppStateStatus) => {
+            const prevAppState = appStateRef.current;
+
+            if (prevAppState.match(/inactive|background/) && nextAppState === 'active') {
+                checkPermission();
+                checkAccuracyAuthorization();
+            }
+
+            appStateRef.current = nextAppState;
+        };
+
+        // check authorizations when app starts
+        checkPermission();
+        checkAccuracyAuthorization();
+        // keep listening to any location authorizations modification while using app
+        const appStateListener = AppState.addEventListener('change', handleAppStateChange);
+        return () => appStateListener.remove();
+    }, []);
+
     /* location provider status update */
     useEffect(() => {
         if (Platform.OS === 'android') {
-            // update location status when app starts
-            const updateLocationStatus = async () => {
-                const locationStatus: LocationProvidersStatus = await getAndroidLocationProvidersStatus();
-                setLocationProviderStatus(locationStatus);
+            // update location providers status when app starts
+            const updateProvidersLocationStatus = async () => {
+                const locationProvidersStatus: LocationProvidersStatus = await Geolocation.getAndroidLocationProvidersStatus();
+                setLocationProviderStatus(locationProvidersStatus);
             };
-            updateLocationStatus().catch(console.log);
+            updateProvidersLocationStatus().catch(console.log);
 
-            // keep listening to any location settings modification while using app
-            const locationProviderListener = locationEmitter.addListener('Location_check', (payload: LocationProvidersStatus) => {
-                setLocationProviderStatus(payload);
-            });
+            // keep listening to any location providers status while using app
+            const locationProviderListener = Geolocation.locationEmitter.addListener(
+                'Location_check',
+                (payload: LocationProvidersStatus) => {
+                    setLocationProviderStatus(payload);
+                }
+            );
             return () => locationProviderListener.remove();
         }
+        return () => null;
```

Comment:
I don't think you need this line

Reply:
for linter

Reply:
Oh ok, maybe weird config of the linter, it shouldn't be needed, but ok

Reply:
I will check